projects/plugins/jetpack/extensions/blocks/math/edit.js

```diff
@@ -0,0 +1,103 @@
+import { useBlockProps } from '@wordpress/block-editor';
+import {
+	Button,
+	TextareaControl,
+	/* eslint-disable @wordpress/no-unsafe-wp-apis */
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	__experimentalText as Text,
+	/* eslint-enable @wordpress/no-unsafe-wp-apis */
+	Placeholder,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { useState, useEffect, useRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { icon } from './icon';
+import 'katex/dist/katex.min.css';
+import './editor.scss';
+import { renderMath } from './utils';
+
+const LatexPlaceholder = () => {
+	return (
+		<span className="jetpack-math-rendered-placeholder">{ __( 'Write Math…', 'jetpack' ) }</span>
+	);
+};
+
+export default function Edit( { attributes, setAttributes, isSelected } ) {
+	const { source = '' } = attributes;
+	const textareaRef = useRef();
+	const [ renderAreaRef, setRenderAreaRef ] = useState( null );
+	const dispatch = useDispatch();
+	const blockProps = useBlockProps();
+	const { style, ...props } = blockProps;
+
+	useEffect( () => {
+		if ( renderAreaRef ) {
+			renderAreaRef.innerHTML = renderMath( source );
+		}
+	}, [ source, renderAreaRef ] );
+
+	if ( isSelected ) {
+		return (
+			<div { ...props }>
+				<Placeholder label={ __( 'Math', 'jetpack' ) } icon={ icon }>
+					<VStack
+						className="jetpack-math-textarea-container"
+						gap={ 4 }
+						justify="flex-start"
+						align="stretch"
+					>
+						<TextareaControl
+							label={ __( 'Math code (Tex or MathML)', 'jetpack' ) }
+							className="jetpack-math-textarea"
+							ref={ textareaRef }
+							value={ source }
+							onChange={ value => setAttributes( { source: value } ) }
+							rows={ 4 }
+							help={ __( 'To render MathML your code must be wrapped in <math> tags.', 'jetpack' ) }
+						/>
+						<Text upperCase>{ __( 'Preview', 'jetpack' ) }</Text>
+						{ source.trim() && (
+							<div
+								className="jetpack-math-rendered"
+								style={ style }
+								ref={ r => r !== renderAreaRef && setRenderAreaRef( r ) }
+							/>
+						) }
```

Comment:
```suggestion
						{ source.trim() && (
							<>
								<Text upperCase>{ __( 'Preview', 'jetpack' ) }</Text>
								<div
									className="jetpack-math-rendered"
									style={ style }
									ref={ r => r !== renderAreaRef && setRenderAreaRef( r ) }
								/>
							</>
						) }
```

I think it looks better hiding the "Preview" until there's some content:

<img width="619" height="494" alt="Screenshot 2025-08-29 at 10 23 07" src="https://github.com/user-attachments/assets/7a02107e-3d8b-49c2-bc5b-9a9d9d64345f" />
<img width="638" height="370" alt="Screenshot 2025-08-29 at 10 23 26" src="https://github.com/user-attachments/assets/7f361291-e4fd-4cd0-973e-1e417cc55b2e" />